hpx/runtime/applier/apply_helper.hpp

```diff
@@ -116,39 +115,78 @@ namespace hpx { namespace applier { namespace detail
         }
     };
 
+    ///////////////////////////////////////////////////////////////////////////
+    HPX_FORCEINLINE bool needs_to_execute_asynchronously()
```

Comment:
Which criteria are you thinking about? Couldn't we just have a defaulted (to 8 * HPX_THREADS_STACK_OVERHEAD) argument? What else could be different?